vm/systemSmartContracts/oneShardSystemVMEEI.go

```diff
@@ -8,27 +8,28 @@ import (
 	"github.com/multiversx/mx-chain-go/vm"
 )
 
-type systemVMEEI struct {
+type oneShardSystemVMEEI struct {
 	vm.ContextHandler
 }
 
-// NewSystemVMEEI creates a new system vm eei context, mainly used for now in sovereign chain
-func NewSystemVMEEI(vmContext vm.ContextHandler) (*systemVMEEI, error) {
+// NewOneShardSystemVMEEI creates a new system vm eei context, mainly used for now in sovereign chain
+func NewOneShardSystemVMEEI(vmContext vm.ContextHandler) (*oneShardSystemVMEEI, error) {
 	if check.IfNil(vmContext) {
 		return nil, vm.ErrNilSystemEnvironmentInterface
 	}
 
-	return &systemVMEEI{
+	return &oneShardSystemVMEEI{
 		vmContext,
 	}, nil
 }
 
 // SendGlobalSettingToAll handles sending global settings information
-func (sovHost *systemVMEEI) SendGlobalSettingToAll(sender []byte, input []byte) error {
+func (sovHost *oneShardSystemVMEEI) SendGlobalSettingToAll(sender []byte, input []byte) error {
+	// TODO: MX-15470 remove sameShard check from ProcessBuiltInFunction
```

Comment:
actually have your own definition of ProcessBuiltInFunction - do not touch the ProcessBuiltInFunction from the original EEI.